### PR TITLE
feat: styling options + add tamagui as a styling option

### PR DESCRIPTION
**Pull request by alitnk**
WIP

### CODE REVIEW

**Comment by danstepanov** [comment]
this is so exciting!

**Comment by alitnk** [comment]
@danstepanov Can you review the PR? Having trouble testing it properly (`yarn test` gets stuck it seems)

**Comment by danstepanov** [comment]
@alitnk reviewing this today

File: src/templates/base/babel.config.js.ejs [hunk 1]
```diff
@@ -2,10 +2,20 @@ module.exports = function(api) {
   api.cache(true);
   let plugins = [];
   
-  <% if (props.useNativewind) { %>
+  <% if (props.stylingPackage?.name === "nativewind") { %>
     plugins.push('nativewind/babel');
   <% } %>
 
+  <% if (props.stylingPackage?.name === "nativewind") { %>
```
Comment (danstepanov) [inline src/templates/base/babel.config.js.ejs:9]: @alitnk did you mean this to be 
```suggestion
  <% if (props.stylingPackage?.name === "tamagui") { %>
```

File: src/templates/packages/tamagui/App.tsx.ejs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import "react-native-gesture-handler";
+import { TamaguiProvider } from 'tamagui'
+import RootStack from "./src/navigation";
```
Comment (danstepanov) [inline src/templates/packages/tamagui/App.tsx.ejs:3]: @alitnk this seems like it wouldn't work if you used Expo Router and Tamagui, no?

**Comment by danstepanov** [comment]
@alitnk this PR seems to be broken down into two chunks:

(1) add tamagui support via a new package
(2) change all instances of `useNativewind` to `props.stylingPackage?.name === "nativewind"`

Can you make this PR have 2 commits that convey that? I rebased the branch with `upstream main`.

File: src/templates/packages/tamagui/App.tsx.ejs (hunk 2 above)
Comment (alitnk) [reply src/templates/packages/tamagui/App.tsx.ejs]: 24383898617f83699cba1aa3e90f67092c3ba804

**Review by danstepanov — APPROVED** [review]
Going to test after merging to `main`

**Pull request merged**